opennms-provision/opennms-provision-persistence/src/main/java/org/opennms/netmgt/provision/persist/policies/ScriptPolicy.java

```diff
@@ -121,13 +130,31 @@ public OnmsNode act(final OnmsNode node) {
             if (compiledScript == null) {
                 LOG.warn("No compiled script available for execution.");
             } else {
-                final SimpleBindings globals = new SimpleBindings();
-                globals.put("node", node);
-                globals.put("LOG", LOG);
-                return (OnmsNode) compiledScript.eval(globals);
+                return sessionUtils.withTransaction(() -> {
+                    try {
+                        // When using sessionUtils (programmatic), the object coming from @Transactional (declarative)
```

Comment:
The problem here is, that the provided `node` is detached (it left the session scope at one point) and when opening the session context again (either via `@Transactional` or `sessionUtils.withTransaction(() -> {})` it is unknown. In order to use it properly it must be re-attached (not that easy with plain hibernate) or simply refetched from the database.